src/multiinstances/internal/ipc/ipcsocket.cpp

```diff
@@ -57,6 +57,10 @@ bool IpcSocket::connect(const QString& serverName)
             }
         });
 
+        QObject::connect(m_socket, &QLocalSocket::disconnected, [this]() {
+            m_disconnected.notify();
```

Comment:
Maybe keep only this solution and remove with PeerClosedError ?

Reply:
fixed